### PR TITLE
move assemble function out of plugin to new file

### PR DESCRIPTION
**Pull request by weizman**

### CODE REVIEW

File: packages/scorchwrap/src/plugin.js [hunk 1]
```diff
@@ -8,18 +8,13 @@
...
-  RuntimeModule,
-  // RuntimeGlobals,
-  // ModuleDependency,
-} = require('webpack')
+const { WebpackError } = require('webpack')
 const { wrapper } = require('./buildtime/wrapper')
 const { generateIdentifierLookup } = require('./buildtime/aa')
 const diag = require('./buildtime/diagnostics')
 const progress = require('./buildtime/progress')
+const { assembleRuntime } = require('./buildtime/assemble')
```
Comment (naugtur) [inline packages/scorchwrap/src/plugin.js:16]: (nit) I'd name the file assembleRuntime or runtimeBuilder for immediate clarity that it's not involved in processing the app sources in any way. But I'm not sure if a person who knows what runtime means in the contect of webpack would not consider it obvious by that point.

File: packages/scorchwrap/src/buildtime/assemble.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        LAVAMOAT['${name}'] = module.exports;
+      })();`
+    }
+    assembly += `\n;/*${name}*/;\n${sourceString}`
+  })
+  assembly += `;
+  __webpack_require__.${KEY} = LAVAMOAT.defaultExport;
+  (typeof harden !== 'undefined') && harden(__webpack_require__.${KEY});` // The harden line is likely unnecessary as the handler is being frozen anyway
+  return {
+    addTo({ compilation, chunk }) {
```
Comment (naugtur) [inline packages/scorchwrap/src/buildtime/assemble.js]: (could) we could refactor this part while we're at it and make this module return a VirtualRuntimeModule that then gets put where it belongs by the code in plugin.js
I regret creating the addTo function now. I used to add multiple runtime modules but that's no longer the case

File: packages/scorchwrap/src/plugin.js (hunk 1 above)
Comment (weizman) [reply packages/scorchwrap/src/plugin.js:16]: how about "build"? I like keeping files to a single word when possible, not by any means of course

File: packages/scorchwrap/src/buildtime/assemble.js (hunk 2 above)
Comment (weizman) [reply packages/scorchwrap/src/buildtime/assemble.js]: maybe something like 6fa51b506?

**Review by naugtur — APPROVED** [review]

**Comment by naugtur** [comment]
Please clean up history before merging. I've rebased my branch to have clean commits there which makes this more difficult. But if you intended to squash, it should be fairly easy to rebase or cherry-pick on top.

**Pull request merged**